src/piece_picker.cpp

```diff
@@ -367,6 +367,11 @@ namespace libtorrent {
 		*zero_prio = int(m_downloads[piece_pos::piece_zero_prio].size());
 	}
 
+    void piece_picker::set_sequential_start(piece_index_t const piece)
+    {
+        sequential_start = piece;
```

Comment:
there should be a bounds-check to ensure this always is set to a valid piece index

Reply:
If it's not the right piece index, will the download just stop? There will be no error?
I don't think we need to add a check here.

Reply:
whatever behavior you think is the most reasonable should be documented.

Reply:
Let me leave the documentation to you. I have a different style of documentation. I don't want to spoil your style of documentation that you already have.

Besides, I don't really understand what needs to be documented.